common/djangoapps/entitlements/models.py

```diff
@@ -362,23 +359,18 @@ def get_fulfillable_entitlement_for_user_course_run(cls, user, course_run_key):
             course_run_key (CourseKey): The course run Key.
 
         Returns:
-            CourseEntitlement: The most recent fullfillale CourseEntitlement, None otherwise.
+            CourseEntitlement: The most recent fulfillable CourseEntitlement, None otherwise.
         """
         # Check if the User has any fulfillable entitlements.
-        # Note: Wait to retrieve the Course UUID until we have confirmed the User has fulfullable entitlements.
+        # Note: Wait to retrieve the Course UUID until we have confirmed the User has fulfillable entitlements.
         # This was done to avoid calling the APIs when the User does not have an entitlement.
         entitlements = cls.get_fulfillable_entitlements(user)
         if entitlements:
             course_uuid = get_course_uuid_for_course(course_run_key)
             if course_uuid:
-                for entitlement in entitlements:
-                    if course_uuid == entitlement.course_uuid:
-                        # Verify that the enrollment can be enrolled in
-                        if is_course_run_entitlement_fulfillable(
-                                course_run_key=course_run_key,
-                                entitlement=entitlement
-                        ):
-                            return entitlement
+                entitlement = entitlements.filter(course_uuid=course_uuid).order_by('-created').first()
```

Comment:
you should only be ordering by '-created' in one place. It might be more appropriate to do it here than in `get_fulfillable_entitlements` for clarity of what is intended to be selected as the fulfillable entitlement for user the given user and course run, but in that case I would remove the ordering on the queryset above in `get_fulfillable_entitlements`.